aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/StateMachine.kt

```diff
@@ -28,9 +28,15 @@ typealias StateChangeListenerToken = UUID
 typealias OnSubscribedCallback = () -> Unit
 
 /**
- * State Machine operates on data/enum classes of State implementation and
- * Resolvers of type State.
+ * Models, mutates and processes effects of a system as a finite state automaton. It consists of:
+ * State - which represents the current state of the system
+ * Resolver - a mechanism for mutating state in response to events and returning side effects called Actions
+ * Listener - which accepts and enqueues incoming events
+ * StateChangedListeners - which are notified whenever the state changes
+ * EffectExecutor - which resolves and executes side Effects/Actions
  * @implements EventDispatcher
+ * @param resolver responsible for mutating state based on incoming events
+ * @param environment holds system specific environment info accessible to Effects/Actions
```

Comment:
Do we need comments describing the other parameters (executor, concurrentQueue, initialState) as well?